File: apis/src/main/java/com/beat/domain/performance/application/dto/performanceDetail/PerformanceDetailCastResponse.java
package com.beat.domain.performance.application.dto.performanceDetail;

public record PerformanceDetailCastResponse(
	Long castId,
	String castName,
	String castRole,
	String castPhoto
) {
	public static PerformanceDetailCastResponse of(Long castId, String castName, String castRole, String castPhoto) {
		return new PerformanceDetailCastResponse(castId, castName, castRole, castPhoto);
	}
}

File: apps/api/src/main/kotlin/com/beat/apps/api/performance/api/response/PerformanceDetailCastResponse.kt
package com.beat.apps.api.performance.api.response

import com.beat.application.frontoffice.performance.CastResult
import com.beat.apps.api.web.jackson.CdnImageUrl

@ConsistentCopyVisibility
data class PerformanceDetailCastResponse
private constructor(
    val castId: Long?,
    val castName: String?,
    val castRole: String?,
    @field:CdnImageUrl val castPhoto: String?,
) {
    companion object {
        fun from(result: CastResult): PerformanceDetailCastResponse =
            PerformanceDetailCastResponse(result.id, result.name, result.role, result.photo)
    }
}
